test/jdk/javax/swing/text/html/bug8314731.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8314731
+ * @summary FormView doesn't support the alt attribute
+ * @library /java/swing/text/html
+*/
+
+import java.awt.BorderLayout;
+import java.awt.Component;
+import java.awt.Container;
+import java.awt.Dimension;
+import java.lang.reflect.InvocationTargetException;
+
+import javax.swing.JButton;
+import javax.swing.JEditorPane;
+import javax.swing.JFrame;
+import javax.swing.JScrollPane;
+import javax.swing.SwingUtilities;
+import javax.swing.text.Document;
+import javax.swing.text.html.HTMLEditorKit;
+import javax.swing.text.html.StyleSheet;
+
+public class bug8314731 {
+
+    private static boolean testFailed = true;
+
+    public static void main(String[] args) throws Exception {
+        new bug8314731();
```

Comment:
the UI needs to be created on the EDT using SwingUtilities.invokeAndWait.
And you need use a finally block to be SURE the Frame is disposed, else the test will just sit there.

Please make sure you have used jtreg to validate this test.

Reply:
Test Passed with JDK master patched
<details>

```
----------System.out:(1/3)----------
ok
----------System.err:(1/15)----------
STATUS:Passed.
----------rerun:(40/2653)*----------
cd /home/scientificwaredev/Documents/openjdk-sources/JTwork/scratch && \\
DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/1000/bus \\
DESKTOP_SESSION=01plasma \\
DISPLAY=:0 \\
HOME=/home/scientificwaredev \\
LANG=fr_FR.UTF-8 \\
PATH=/bin:/usr/bin:/usr/sbin \\
XDG_CONFIG_DIRS=/home/scientificwaredev/.config/kdedefaults:/etc/xdg:/etc/xdg/kf5:/var/lib/plasma5-profiles/common/xdg \\
XDG_CURRENT_DESKTOP=KDE \\
XDG_DATA_DIRS=/home/scientificwaredev/.local/share/flatpak/exports/share:/var/lib/flatpak/exports/share:/usr/local/share:/usr/share \\
XDG_MENU_PREFIX=plasma- \\
XDG_RUNTIME_DIR=/run/user/1000 \\
XDG_SEAT=seat0 \\
XDG_SEAT_PATH=/org/freedesktop/DisplayManager/Seat0 \\
XDG_SESSION_CLASS=user \\
XDG_SESSION_DESKTOP=KDE \\
XDG_SESSION_ID=c2 \\
XDG_SESSION_PATH=/org/freedesktop/DisplayManager/Session1 \\
XDG_SESSION_TYPE=x11 \\
XDG_VTNR=2 \\
XMODIFIERS=@im=none \\
CLASSPATH=/home/scientificwaredev/Documents/openjdk-sources/JTwork/classes/javax/swing/text/html/bug8314731.d:/home/scientificwaredev/Documents/openjdk-sources/jdk/test/jdk/javax/swing/text/html:/home/scientificwaredev/Documents/jtreg/lib/javatest.jar:/home/scientificwaredev/Documents/jtreg/lib/jtreg.jar \\
    /home/scientificwaredev/Documents/openjdk-sources/jdk/build/linux-x86_64-server-release/jdk/bin/java \\
        -Dtest.vm.opts= \\
        -Dtest.tool.vm.opts= \\
        -Dtest.compiler.opts= \\
        -Dtest.java.opts= \\
        -Dtest.jdk=/home/scientificwaredev/Documents/openjdk-sources/jdk/build/linux-x86_64-server-release/jdk \\
        -Dcompile.jdk=/home/scientificwaredev/Documents/openjdk-sources/jdk/build/linux-x86_64-server-release/jdk \\
        -Dtest.timeout.factor=1.0 \\
        -Dtest.root=/home/scientificwaredev/Documents/openjdk-sources/jdk/test/jdk \\
        -Dtest.name=javax/swing/text/html/bug8314731.java \\
        -Dtest.file=/home/scientificwaredev/Documents/openjdk-sources/jdk/test/jdk/javax/swing/text/html/bug8314731.java \\
        -Dtest.src=/home/scientificwaredev/Documents/openjdk-sources/jdk/test/jdk/javax/swing/text/html \\
        -Dtest.src.path=/home/scientificwaredev/Documents/openjdk-sources/jdk/test/jdk/javax/swing/text/html \\
        -Dtest.classes=/home/scientificwaredev/Documents/openjdk-sources/JTwork/classes/javax/swing/text/html/bug8314731.d \\
        -Dtest.class.path=/home/scientificwaredev/Documents/openjdk-sources/JTwork/classes/javax/swing/text/html/bug8314731.d \\
        -Dtest.modules=java.desktop \\
        --add-modules java.desktop \\
        com.sun.javatest.regtest.agent.MainWrapper /home/scientificwaredev/Documents/openjdk-sources/JTwork/javax/swing/text/html/bug8314731.d/main.0.jta
result: Passed. Execution successful


test result: Passed. Execution successful

```
</details>

Test Failed with JDK 23.0.1.
<details>

```
----------System.out:(0/0)----------
----------System.err:(11/728)----------
java.lang.RuntimeException: FormView doesn't support the alt attribute, see JDK-8314731.
	at bug8314731.main(bug8314731.java:57)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at com.sun.javatest.regtest.agent.MainWrapper$MainTask.run(MainWrapper.java:138)
	at java.base/java.lang.Thread.run(Thread.java:1575)

JavaTest Message: Test threw exception: java.lang.RuntimeException: FormView doesn't support the alt attribute, see JDK-8314731.
JavaTest Message: shutting down test

STATUS:Failed.`main' threw exception: java.lang.RuntimeException: FormView doesn't support the alt attribute, see JDK-8314731.
----------rerun:(40/2512)*----------
cd /home/scientificwaredev/Documents/openjdk-sources/JTwork/scratch && \\
DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/1000/bus \\
DESKTOP_SESSION=01plasma \\
DISPLAY=:0 \\
HOME=/home/scientificwaredev \\
LANG=fr_FR.UTF-8 \\
PATH=/bin:/usr/bin:/usr/sbin \\
XDG_CONFIG_DIRS=/home/scientificwaredev/.config/kdedefaults:/etc/xdg:/etc/xdg/kf5:/var/lib/plasma5-profiles/common/xdg \\
XDG_CURRENT_DESKTOP=KDE \\
XDG_DATA_DIRS=/home/scientificwaredev/.local/share/flatpak/exports/share:/var/lib/flatpak/exports/share:/usr/local/share:/usr/share \\
XDG_MENU_PREFIX=plasma- \\
XDG_RUNTIME_DIR=/run/user/1000 \\
XDG_SEAT=seat0 \\
XDG_SEAT_PATH=/org/freedesktop/DisplayManager/Seat0 \\
XDG_SESSION_CLASS=user \\
XDG_SESSION_DESKTOP=KDE \\
XDG_SESSION_ID=c2 \\
XDG_SESSION_PATH=/org/freedesktop/DisplayManager/Session1 \\
XDG_SESSION_TYPE=x11 \\
XDG_VTNR=2 \\
XMODIFIERS=@im=none \\
CLASSPATH=/home/scientificwaredev/Documents/openjdk-sources/JTwork/classes/javax/swing/text/html/bug8314731.d:/home/scientificwaredev/Documents/openjdk-sources/jdk/test/jdk/javax/swing/text/html:/home/scientificwaredev/Documents/jtreg/lib/javatest.jar:/home/scientificwaredev/Documents/jtreg/lib/jtreg.jar \\
    /home/scientificwaredev/Documents/jdk-23.0.1/bin/java \\
        -Dtest.vm.opts= \\
        -Dtest.tool.vm.opts= \\
        -Dtest.compiler.opts= \\
        -Dtest.java.opts= \\
        -Dtest.jdk=/home/scientificwaredev/Documents/jdk-23.0.1 \\
        -Dcompile.jdk=/home/scientificwaredev/Documents/jdk-23.0.1 \\
        -Dtest.timeout.factor=1.0 \\
        -Dtest.root=/home/scientificwaredev/Documents/openjdk-sources/jdk/test/jdk \\
        -Dtest.name=javax/swing/text/html/bug8314731.java \\
        -Dtest.file=/home/scientificwaredev/Documents/openjdk-sources/jdk/test/jdk/javax/swing/text/html/bug8314731.java \\
        -Dtest.src=/home/scientificwaredev/Documents/openjdk-sources/jdk/test/jdk/javax/swing/text/html \\
        -Dtest.src.path=/home/scientificwaredev/Documents/openjdk-sources/jdk/test/jdk/javax/swing/text/html \\
        -Dtest.classes=/home/scientificwaredev/Documents/openjdk-sources/JTwork/classes/javax/swing/text/html/bug8314731.d \\
        -Dtest.class.path=/home/scientificwaredev/Documents/openjdk-sources/JTwork/classes/javax/swing/text/html/bug8314731.d \\
        -Dtest.modules=java.desktop \\
        --add-modules java.desktop \\
        com.sun.javatest.regtest.agent.MainWrapper /home/scientificwaredev/Documents/openjdk-sources/JTwork/javax/swing/text/html/bug8314731.d/main.0.jta
result: Failed. Execution failed: `main' threw exception: java.lang.RuntimeException: FormView doesn't support the alt attribute, see JDK-8314731.


test result: Failed. Execution failed: `main' threw exception: java.lang.RuntimeException: FormView doesn't support the alt attribute, see JDK-8314731.

```
</details>